salt/ca/init.sls

```diff
@@ -54,11 +54,7 @@ salt-minion:
 /etc/pki/ca.crt:
   x509.certificate_managed:
     - signing_private_key: /etc/pki/ca.key
-{% if salt['grains.get']('domain', None) is none %}
-    - CN: {{ pillar['dns']['domain'] }}
-{% else %}
-    - CN: {{ grains['domain'] }}
-{% endif %}
+    - CN: {{ salt['grains.get']('domain', pillar['dns']['domain']) }}
```

Comment:
No, `pillar['dns']['domain']` shouln't be empty, ever. What I think is that `grains['domain']` instead of being `None` is `""`, and the default value is not applied, thus `CN: ""`, failing again.